parser_library/src/checking/data_definition/data_def_type_base.h

```diff
@@ -18,12 +18,17 @@
 #include <cstdint>
 #include <functional>
 #include <map>
+#include <memory>
```

Comment:
Probably not needed

```suggestion
```

Reply:
It failed to compile without it (the unique_ptr is actually used below), so it must have previously been included by accident through a different header.

Reply:
It must be still included from somewhere as it does indeed compile (at least with msvc).

Reply:
worked with msvc and clang with libc++, failed on gcc/stdlibc++.